lib/api/decorators/collection.rb

```diff
@@ -47,7 +47,19 @@ def self.element_decorator(klass)
       end
 
       def element_decorator
-        self.class.element_decorator_class
+        self.class.element_decorator_class || deduce_element_decorator
+      end
+
+      def deduce_element_decorator
+        name = self.class.name
+
+        unless name.end_with?('CollectionRepresenter')
+          raise ArgumentError, "Can't deduce representer name from #{name}, please specify it with `element_decorator ClassName`"
+        end
+
+        name
+          .gsub("CollectionRepresenter", "Representer")
+          .constantize
```

Comment:
:+1: I like that. 

We might to eventually generate some of the boilerplate classes, e.g.:
* CollectionRepresenter
* PayloadRepresenter
* CreateService
* UpdateService
* DeleteService

automatically if they do not exist. That way, if they behave like the superclass, nothing needs to be done and one does not feel bad if no test is written for them. That would focus the work even more on where the knowledge is.